### ISSUE TITLE
ISSkiOM 2019 symbol set

### ISSUE THREAD

**Issue by Symbian9** [OpenOrienteering/mapper#1408]
*International Specification for Ski Orienteering Maps* (*ISSkiOM 2019* **valid** from 1st November 2019) is now available online [on *IOF website*](https://orienteering.sport/iof/mapping/). (old specs still available [in *MC Archive*](https://orienteering.sport/iof/mapping/mc-archive/))

It approved on October 24-25, 2019 (according [*IOF Counsil Meeting Minutes* `#196`](https://orienteering.sport/iof/governance-and-organisation/council/)), so it's valid already 7 days.

There are two PDFs available:

- `ISSkiOM 2019 (valid from 1st November 2019).pdf` (spec)
- `Ski-O map symbols 2019.pdf`  (map sample)
  -  **UPD:** IOF MC already corrected and re-uploded map sample on November 11, 2019.

But Finnish mapper *Pasi Jokelainen* reported that map sample ~~has~~ had issues:

> A little comment from Finland:
>
> There is an error in the model map. The yellow map symbols still contain old map symbols. Map symbols 402 and 404 are a bit wrong now.
>
> (the model map is familiar to me since we did it 20 years ago).

**Comment by dg0yt** [comment]
> ![facebook](https://user-images.githubusercontent.com/13567791/68459015-7f794480-0204-11ea-9e50-7aca7f07a7ff.png)

# Disclaimer
**The presence of this feature request does not indicate that this feature is coming soon, even if some people claim so on Facebook.** 

This is really one posting too much now. While I appreciate actual news being published on Facebook, I strongly dislike the misrepresentation given by
- announcing coming features which noone even claimed to work on,
- sometimes giving misleading support on technical problems, due to lack on insight into the technical background,
- claiming that www.openorienteering.org is the homepage of a "OpenOrienteering Ukraine" Facebook organization.

I will lock conversation in this issue for some time. Take a break.

Kai Pastor, DG0YT 
OpenOrienteering developer

**Comment by dg0yt** [comment]
@krticka Issues with ISSkiOM 2019
- [ ] **The layer position of the green color for track symbols is undefined.**
  We can make an informed guess, but the MC gave the impression that they want to standardize how colors are implemented.
- [ ] **Fulltone green vs. dark green, colors vs symbols.**
  When converting ISOM (2017-2) maps, it is required that the "dark green (symbol 410) is replaced by light green (symbol 406)".
  Shall 406 keep the name "Vegetation: slow running" if it used for areas which are in fact "Vegetation: fight"?
  I don't think that "dark green" and "light green" should be used here. Symbol 410 is not dark green. Even with the extra "Green + 50% Black" from ISSprOM and ISOM 2017 removed as an area symbol in ISOM 2017-2, *this dark green* is still in use ISOM 2017-2 for 416 "Distinct vegetation boundary". 
  "All vegetation screens must be drawn with the symbol Vegetation: slow running (406)." Again , this also mixes color terminology (screen) with symbol terminonlogy. How does it apply to 402 "Open land with scattered trees" which may use a Green 60% screen?
  And with 410 replaced by 406, what about 408? Shall an ISSkiOM-conforming map file use *code number 408* with the same "screen" as 406, or shall these features use *code number 406*?

- [ ] **The "Other symbols" use non-integer code numbers.**
  These numbers create conflicts with numbers which we used to implement regular ISOM 2017-2 symbols.
  I already raised this code number issue for ISSprOM 2019. In the past, the standards used only integer numbers. Subnumbers were left for implementation aspects. It is disappointing to see this again in ISSkiOM, only few days after we integrated an ISOM 2017-2 symbol set. We could have avoided that conflict so easily if we had been able to know what is coming.
  However, the "Course planning symbols" such as 70 Control point use a different graphical appearance without using subnumbers.
- [ ] **The code number of "Crossable body of water" is a variant of the ISOM code number for "Uncrossable body of water" (and also a close sibling of the ISSkiOM "Uncrossable body of water").** 
 I acknowledge that there is no "Crossable body of water" in ISOM, but at least there is "Shallow body of water" as a contrast to uncrossable ones. I feel that the increasing inconsistency of the meaning of code numbers over time and map standards has reduced their usability as ... *code numbers*.
- [ ] **The standard became effective before being published.**
  Thus it is impossible to implement map standards in time, let alone to give feedback on issues found during implementation. I doubt that perspectives like formal consistency, usability with mapping software, upgrading of older maps, switching from ISOM 2017-2 to ISSkiOM, and software interoperability can be expected to be sufficiently covered by feedback from IOF member societies. At least this is my impression with ISSOM 2017, ISSOM 2017-2, ISSpOM (pending in Mapper) and now ISSkiOM.

**Comment by Symbian9** [comment]
IOF MC just released update to Appendix  1 with colors order for *ISSkiOM 2019*'s additional colors.

- `ISOM 2017 Appendix 1 CMYK Printing and Colour Definitions (temporary including ISprOM) Version 3.pdf`

Also IOF website now available symbol sets  *ISSkiOM 2019* in OCDv12 and OCD2019 formats.

Additionally, `Ski-O map symbols 2019.pdf` map sample corrected.

**Comment by krticka** [comment]
> The layer position of the green color for track symbols is undefined.

The position is quite clear from Ski-O maps document, the colour order is as follows:
- Upper purple
- White (rub out all)
- Green for Ski-O
- Black 100%
- Brown 100%
- Lower purple

> Fulltone green vs. dark green, colors vs symbols.

I agree that terminology used is not consistent here and text could be written more precisely. List of symbols in italics clearly defines what symbols from ISOM 2017-2 are used in ISSkiOM.
Specification says: "In order not to destroy legibility of the green tracks, all **vegetation** screens must be drawn with the symbol Vegetation: slow running (406)". In case of this specification it always means symbols for **vegetation reducing running** (408 Vegetation: walk, 410 Vegetation: fight). 

> Shall an ISSkiOM-conforming map file use code number 408 with the same "screen" as 406, or shall these features use code number 406?

That's the technical question which I think is out of scope of specification. In my opinion, thinking that Ski-O maps are usually done by converting foot-O I think it is more practical to do conversion of 410 and 408 to 406 as having the same look for different symbols would be confusing for users. Also, when the Ski-O map will be created from the beginning, it is better solution to have 406 only.

>The "Other symbols" use non-integer code numbers.

The map specification nomenclature has here its logic to use non-integer values as these are special variants of ISOM symbols used in Ski orienteering. I don't think that putting these symbols under completely different integer number is wise for the users. I understand that this is complication now as these numbers were already used in ISOM 2017-2 legend, but announce and predict possible problems in advance is sometimes hard to achieve (world is not perfect).  Simply it is task for software developers to deal with this, for example by redefining the ISOM symbol codes or use some type of sub coding for Ski-O (301.1.1?).  

> However, the "Course planning symbols" such as 70 Control point use a different graphical appearance without using subnumbers

If I understand your note correctly (different position of purple colour), yes, there is different graphical appearance from ISOM as Ski-O uses similar setting as MTBO. We had some discussions with Ski-O Commission about this as in the past the use of purple symbols and position of purple was not clear for organizers. 

> The code number of "Crossable body of water" is a variant of the ISOM code number for "Uncrossable body of water" (and also a close sibling of the ISSkiOM "Uncrossable body of water").

In Ski-O crossability of water features is something different than in ISOM and ISSprOM. In winter lakes are frozen and can be used is competition, thus are crossable. However there can be fluctuations in weather and crossing can be dangerous, thus you need to change these water features to Body of water that shall not be crossed. If the lakes will be used/will not be used is usually decided by organizer in very short time before competition. In other words, when converting ISOM map, it is not clear in advance, which variant will organizer use. I would prefer to convert ISOM 301 to ISSkiOM 301.2.

> The standard became effective before being published.

Not MC decision. Ski-O Commission wanted to have this for upcoming season and Council had to approve it on its regular meeting. 

In reality if you look what the changes are to the symbol part, the amount of changes is very low comparing to previous version of ISSkiOM. I hope that I was able to help you and clarify the problematic parts. I will distribute your comments to the other MC members to have some feedback on specification.

**Comment by krticka** [comment]
> 
> 
> On IOF website now available _ISSkiOM 2019_ symbol set in OCDv12 and OCD2019 formats.
> 
> Additionally, `Ski-O map symbols 2019.pdf` map sample corrected.

Please note that Ski-O map symbols 2019 document is absolutely inappropriate for map creation. It is guide for users to understand different scales and symbol sizes used in Ski-O. For this reason it mixes different symbol sizes in legend and colours in colour table.

**Comment by dg0yt** [comment]
> The position is quite clear from Ski-O maps document

I don't think *obvious* is enough when otherwise things are made *explicit*.

> I agree that terminology used is not consistent here and text could be written more precisely. List of symbols in italics clearly defines what symbols from ISOM 2017-2 are used in ISSkiOM.
Specification says: "In order not to destroy legibility of the green tracks, all vegetation screens must be drawn with the symbol Vegetation: slow running (406)". In case of this specification it always means symbols for vegetation reducing running (408 Vegetation: walk, 410 Vegetation: fight).

So no change for 402. (?)

> In my opinion, thinking that Ski-O maps are usually done by converting foot-O I think it is more practical to do conversion of 410 and 408 to 406 as having the same look for different symbols would be confusing for users. Also, when the Ski-O map will be created from the beginning, it is better solution to have 406 only.

I also consider both use cases, converting existing maps, and creating new maps. I want to offer the right colors, right symbols and consistent names. When I only offer 406, I need to give it a different name (and description) than in ISOM, but it is undefined (at the moment) in ISSkiOM. On the other hand, ISSkiOM is explicit about bodies of water.

>  Simply it is task for software developers to deal with this, for example by redefining the ISOM symbol codes or use some type of sub coding for Ski-O (301.1.1?)

SEP, Someone Else's Problem now ...it became a task for software developers and map makers, but it is not "simple".
- I can easily redefine the code numbers in our ISOM symbol set, but this does not apply so easily for map makers who already started using the symbol set.
- I did not use 301.1.1 style until now because it limits interoperability with the OCD format.
- I also easily cannot start with implementation sub-code numbers higher than 9 because it limits interoperability with OCD format version 8. Granted, this is of lower relevance nowadays, but on the other hand some users of our software appreciate that feature.
- I cannot really know if the standards won't come up with a 301.1.1, too. Granted, this is unlikely, not only because...
- It looks ugly.

> In Ski-O crossability of water features is something different than in ISOM and ISSprOM. In winter lakes are frozen and can be used is competition, thus are crossable. However there can be fluctuations in weather and crossing can be dangerous, thus you need to change these water features to Body of water that shall not be crossed. If the lakes will be used/will not be used is usually decided by organizer in very short time before competition. In other words, when converting ISOM map, it is not clear in advance, which variant will organizer use.

Yes, I'm very aware that crossable body of water has a different meaning in the winter, and map makers and organizers must review these objects. But the key aspect is crossability signaled by different symbols, and I think it would make more sense to keep the 301 and 302 code numbers which carry this aspect (301 at least), with the desired variants of graphical representations, and maybe an adjusted name for ISSkiOM 302.

**Comment by Symbian9** [comment]
As *Mapper* latest dev build already include ISSkiOM 2019 symbol set, there is some issue:

- [x] Name in symbol set chooser window: `ISSkiOM_<scale>`  — it's OK, as *Mapper* actually provided with symbol sets according latest IOF specs

![pic.1](https://i.imgur.com/lkimr5w.png)

- [ ] Symbol set ID: `ISSkiOM` — this is wrong, as symbol set ID should really describe what version of spec used for this symbol set.

![pic.2](https://i.imgur.com/3ozuNFc.png)

**Comment by dg0yt** [comment]
@Symbian9 Screenshot and symbol set ID are right, but your conclusion is wrong: Unfortunately, the latest master package still has the old ISSkiOM files. I need to update the release notes, and perhaps add them as separate link or asset.

**Comment by krticka** [comment]
@dg0yt sorry for late reply, I was too busy.

I can see ISSkiOM on the web, good work! Just to be sure, I will try to reply.

> I don't think obvious is enough when otherwise things are made explicit.

Tell me that. I wanted to make the colour order part of updated appendix. It was not my decision not to include it.

> So no change for 402. (?)

No change, Open land with scattered trees is directly used also in ISSkiOM.

> SEP, Someone Else's Problem now ...it became a task for software developers and map makers, but it is not "simple".

Looking at ISSkiOM symbol set I think that you solved it OK.

> Yes, I'm very aware that crossable body of water has a different meaning in the winter, and map makers and organizers must review these objects. But the key aspect is crossability signaled by different symbols, and I think it would make more sense to keep the 301 and 302 code numbers which carry this aspect (301 at least), with the desired variants of graphical representations, and maybe an adjusted name for ISSkiOM 302.

Grooming on frozen lakes is quite common in Scandinavian countries and these areas are used for competition (are crossable). Crossability of lakes can even vary within one map, when there is bigger difference in lake altitude.  Be sure that some organizers are reviewing conditions of the ice. Using 302.1 for Crossable body of water? I would personally still prefer still use some subnumbering of 301 (5, 6), but accept you decision.

**Comment by dg0yt** [comment]
I'm still not happy with the situation for "bodies of water". Look at this comparison (official code numbers underlined), based on screenshots.

![bodies-of-water](https://user-images.githubusercontent.com/13567791/69218052-3b753080-0b70-11ea-8f55-b0641bc0ab67.png)

- Here, sorted by code number, the official crossable symbol moved to the position in front of the uncrossable one.
- Note that I decided to leave the 302 prefix for the components of 301.1. This is why they are not grouped together with 301.1, given this particular sorting.

I had to change (our) 301.1 to 301.3. But I couldn't change (our) 302.1/302.2 to xxx.y numbers which would go in between (ISSkiOM) 301.1/301.2. 

I may add more code to adjust the grouping. But to be honest, I seriously considered to simply ignore the ISSkiOM code numbers, for the benefit of ISOM similarity and for practical consistence. In creating and using maps, the original ISOM numbers would serve all practical purposes as well, except for a purely formal compliance with ISSkiOM. Let me make it explicit:

*301.1 should have remained 302. 
301.2 should have remained 301.*
